iep-008/README.adoc

```diff
@@ -0,0 +1,155 @@
+ifdef::env-github[]
+:tip-caption: :bulb:
+:note-caption: :information_source:
+:important-caption: :heavy_exclamation_mark:
+:caution-caption: :fire:
+:warning-caption: :warning:
+endif::[]
+
+= IEP-008: Ldap on Kubernetes
+
+:toc:
+
+.Metadata
+[cols="2"]
+|===
+| IEP
+| 8
+
+| Title
+| Ldap on Kubernetes
+
+| Author
+| link:https://github.com/olblak[Olblak]
+
+| Status
+| :speech_balloon: In-process
+
+| Type
+| Architecture
+
+| Created
+| 2018-01-31
+|===
+
+
+
+== Abstract
+
+As part of the migration to Azure, the Ldap server must be moved.
+We can take this opportunity to containerize this service and move it on a container orchestrator like the kubernetes cluster.
+This new architecture change must be done carefully as this is a stateful application with a database that can be corrupted or lead to data loss.
+
+== Specification
+
+This IEP document is about running ldap on Kubernetes on Azure.
+Currently the ldap server is running on a bare metal machine and is configured by puppet.
+The objective here, is to dockerize this application in order to run it on a kubernetes cluster.
+
+As a stateful application, we must take into consideration following aspect when deploying the ldap server in production.
+
+=== Access
+Ldap must be accessible from inside and outside the kubernetes cluster.
+In order to reach ldap from the wild, we need a fixed public IP and only allow connection from whitelisted IP.
+This can be easily done with the kubernetes resource 'service'.
+
+[cols="1a,2a", options="header"]
+.Access
+|===
+|Inside
+|Outside
+|
+* https://accounts.jenkins.io/[Accountapp]
+|
+* https://repo.jenkins-ci.org/webapp/#/home[Artifactory]
+* https://ci.jenkins.io[ci.jenkins.io]
+* https://wiki.jenkins.io/[Confluence]
+* https://issues.jenkins-ci.org[Jira]
+* puppet master
+* spambot
+* trusted.ci
+|===
+
+
+=== Backup/Restore
+The procedure to backup/restore the database should be easy, two scripts are provided by the same docker image that run ldap. +
+Backups must be stored on an Azure File Storage in order to simplify their access from various location. +
+Backup name must respect format 'YYYYmmddHHMM'
+
+Backups and restore operation must be done in following situations:
+
+[cols="1a,2a", options="header"]
+.Backup/Restore
+|===
+|Backup
+|Restore
+| * On daily basis
+* When the application is stopping
+* On demand
+| * On demand
+|===
+
+=== Certificate
+I didn't find an easy way to use Letsencrypt certificate from Kube-lego, the constraint that I have at the moment is
```

Comment:
Since the `ldap.jenkins.io` certificate is the only one in the `jenkins.io` domain space to be managed manually, will you please update this to describe that the server's certificate should be coming from Puppet somehow?